tests/integration/nm/linux_bridge_test.py

```diff
@@ -22,24 +22,65 @@
 from libnmstate import nm
 from libnmstate.schema import LinuxBridge as LB
 
-from .testlib import mainloop
+from ..testlib import iproutelib
+from .testlib import mainloop_run
 
 
 BRIDGE0 = 'brtest0'
-ETH1 = 'eth1'
+PORT0 = 'eth1'
+PORT1 = 'eth2'
 
 
 @pytest.fixture
-def bridge_minimum_config():
-    return {
+def bridge0_with_port0(eth1_up):
+    bridge_desired_state = _create_bridge_config((PORT0,))
+    with _bridge_interface(bridge_desired_state):
+        yield _get_bridge_current_state()
+
+
+def test_create_and_remove_minimum_config_bridge():
+    bridge_desired_state = {
         LB.CONFIG_SUBTREE: {
             LB.PORT_SUBTREE: []
         }
     }
 
+    with _bridge_interface(bridge_desired_state):
 
-@pytest.fixture
-def bridge_default_config():
+        bridge_current_state = _get_bridge_current_state()
+        assert bridge_current_state
+        assert not bridge_current_state[LB.CONFIG_SUBTREE][LB.PORT_SUBTREE]
+
+    assert not _get_bridge_current_state()
+
+
+def test_create_and_remove_bridge(eth1_up):
+    bridge_desired_state = _create_bridge_config((PORT0,))
+    with _bridge_interface(bridge_desired_state):
+
+        bridge_current_state = _get_bridge_current_state()
+        assert bridge_desired_state == bridge_current_state
+
+    assert not _get_bridge_current_state()
+
+
+@iproutelib.ip_monitor_assert_stable_link_up(BRIDGE0)
+def test_add_port_to_existing_bridge(bridge0_with_port0, eth2_up):
```

Comment:
Why is this using `eth2_up`?

Reply:
I see, it is because `PORT1` is `eth2` - IMHO this is just too confusing and aliasing the fixtures with the abstract names would make it a lot better.

Reply:
https://github.com/nmstate/nmstate/pull/415